### PR TITLE
update to version 3.3.3

### PR DESCRIPTION
**Pull request by eszlari**

### CODE REVIEW

**Comment by eszlari** [comment]
bot, build

File: krita-appdata-release.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- krita-3.3.3/krita/org.kde.krita.appdata.xml.old	2018-01-08 12:49:24.000000000 +0100
++++ krita-3.3.3/krita/org.kde.krita.appdata.xml	2018-01-10 14:53:12.000000000 +0100
+@@ -179,4 +179,7 @@
+   <provides>
+     <binary>krita</binary>
+   </provides>
++  <releases>
```
Comment (aleixpol) [inline krita-appdata-release.patch]: Send upstream?

Comment (eszlari) [reply krita-appdata-release.patch]: I talked to them, but the tarball was already uploaded. Should be done for the next release.

Comment (eszlari) [reply krita-appdata-release.patch]: Yes, master has it already.

**Comment by eszlari** [comment]
bot, build

**Comment by Pointedstick** [comment]
FWIW, Upstream has an incorrect and invalid AppStream ID (`org.kde.krita.desktop`) that doesn't match the one we have here (`org.kde.krita`).

https://cgit.kde.org/krita.git/tree/krita/org.kde.krita.appdata.xml#n3

I'm gonna submit a patch to fix that soon, if you don't get to it first. :slightly_smiling_face:

**Comment by eszlari** [comment]
That's how it supposed to be named:
https://github.com/search?l=XML&q=org%3Aflathub+desktop&type=Code

But the icons are not named correctly:
https://github.com/flathub/org.kde.krita/blob/master/org.kde.krita.json#L8

**Comment by Pointedstick** [comment]
Everything shown on https://github.com/search?l=XML&q=org%3Aflathub+desktop&type=Code is wrong, sadly--and I have been working to change them. Here's the AppStream spec: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#sect-Metadata-GenericComponent

Relevant excerpt:

> 
>  The <id> tag is a unique identifier for this component. It must contain only ASCII characters, dots, hyphens and numbers. Spaces are not allowed.
> The ID must follow a reverse-DNS scheme, consisting of {tld}.{vendor}.{product}, for example org.kde.Gwenview or com.hugski.ColorHug2. Ownership of {vendor}.{tld} in the domain name system guarantees uniqueness of IDs.

**Comment by eszlari** [comment]
They get validated automatically by appstreamcli after the flatpak build, so how do they pass that?

**Comment by TingPing** [comment]
It isn't *invalid*, I think it is a left over from previous behavior where it was required for the desktop component type. @hughsie is that correct, and moving forward should they be removed and does that cause any breakage?

**Pull request merged**